### PR TITLE
Move the password check for SDC

### PR DESCRIPTION
**Pull request by shekhar-j**
# Description
Value of the parameter, check in validator block fails if parameter is passed as variable eg. var.user. Variable value is not available so it is empty or null. 
Updated package path logic to work on Windows too. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 80% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by doriac11 — APPROVED** [review]

**Review by nikitajoshi1 — APPROVED** [review]

**Review by doriac11 — APPROVED** [review]

**Review by nikitajoshi1 — DISMISSED** [review]

File: powerflex/helper/sdc_host_dpkg.go [hunk 1]
```diff
@@ -84,7 +83,7 @@ func (r *SdcHostResource) CreateUbuntu(ctx context.Context, plan models.SdcHostM
 
 		// upload sw
 		scpProv := client.NewScpProvisioner(sshP)
-		pkgTarget := filepath.Join(dir, "emc-sdc-package.tar")
+		pkgTarget := dir + "/" + "emc-sdc-package.tar"
```
Comment (Krunal-Thakkar) [inline powerflex/helper/sdc_host_dpkg.go:86]: @shekhar-j  Have you tested this on all platforms?

Also, have you run a security check after these changes?

**Comment by shekhar-j** [comment]
run e2e test

**Comment by shekhar-j** [comment]
run e2e test

**Review by nikitajoshi1 — DISMISSED** [review]

File: powerflex/helper/sdc_host_dpkg.go (hunk 1 above)
Comment (rounak-adhikary) [reply powerflex/helper/sdc_host_dpkg.go:86]: To take care of edge case
```suggestion
		pkgTarget := strings.TrimSuffix(dir, "/") + "/" + "emc-sdc-package.tar"
```

File: powerflex/provider/sdc_host_resource.go [hunk 2]
```diff
@@ -95,18 +95,6 @@ func (r *sdcHostResource) ValidateConfig(ctx context.Context, req resource.Valid
 		)
 	}
 
-	if !cfg.OS.IsUnknown() && cfg.OS.ValueString() == "windows" && !cfg.Remote.IsNull() {
```
Comment (rounak-adhikary) [inline powerflex/provider/sdc_host_resource.go:98]: Why is this necessary?

Comment (shekhar-j) [reply powerflex/provider/sdc_host_resource.go:98]: As mentioned in description, this doesn't work if param is passed as var value eg. var.user. So check is moved.

**Review by rounak-adhikary — DISMISSED** [review]

**Review by rounak-adhikary — APPROVED** [review]

**Review by doriac11 — APPROVED** [review]

File: powerflex/helper/sdc_host_dpkg.go (hunk 1 above)
Comment (shekhar-j) [reply powerflex/helper/sdc_host_dpkg.go]: Functional and Acceptance tests for this resource are passing. Security scans are ok.
